src/test/java/core/basesyntax/service/impl/ShopServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.config.AppConfig;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class ShopServiceImplTest {
+    private ShopService shopService;
+    private Map<String, Integer> inventory;
+
+    @BeforeEach
+    void setUp() {
+        AppConfig appConfig = new AppConfig();
+        Map<FruitTransaction.Operation, OperationHandler> operationHandlers =
+                appConfig.getOperationHandlers();
+        shopService = new ShopServiceImpl(operationHandlers);
+        inventory = new HashMap<>();
+    }
```

Comment:
Are you sure you need to create new instances of these classes before each test method? Before all should be enough, also you can clear the inventory if you need it, instead of creating a new HashMap.